pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -35,17 +37,30 @@ func New(adapterContext common.AdapterContext, client kclient.Client) Adapter {
 type Adapter struct {
 	Client kclient.Client
 	common.AdapterContext
+	pod             *corev1.Pod
+	devfileBuildCmd string
+	devfileRunCmd   string
 }
 
 // Push updates the component if a matching component exists or creates one if it doesn't exist
 // Once the component has started, it will sync the source code to it.
-func (a Adapter) Push(path string, ignoredFiles []string, forceBuild bool, globExps []string) (err error) {
+func (a Adapter) Push(path string, ignoredFiles []string, forceBuild bool, show bool, globExps []string, devfileBuildCmd, devfileRunCmd string) (err error) {
 	componentExists := utils.ComponentExists(a.Client, a.ComponentName)
 
+	a.devfileBuildCmd = devfileBuildCmd
+	a.devfileRunCmd = devfileRunCmd
+
 	deletedFiles := []string{}
 	changedFiles := []string{}
 	isForcePush := false
 
+	// Validate the devfile build and run commands
+	pushDevfileCommands, err := common.ValidateAndGetPushDevfileCommands(a.Devfile.Data, a.devfileBuildCmd, a.devfileRunCmd)
+	if err != nil {
+		return err
```

Comment:
Can you wrap this error with a message? Maybe something like "Push failed due to validate error"?